MapboxCoreNavigation/NavigationSettings.swift

```diff
@@ -20,12 +20,31 @@ public class NavigationSettings: NSObject {
      
      This volume is relative to the system’s volume where 1.0 is same volume as the system.
     */
-    @objc public dynamic var voiceVolume: Float     = 1.0
+    @objc public dynamic var voiceVolume: Float = 1.0
     
     /**
-     Indicates whether the voice controller should be muted or not.
+     Specifies whether to mute the voice controller or not.
      */
-    @objc public dynamic var voiceMuted: Bool      = false
+    @objc public dynamic var voiceMuted : Bool = false
+    
+    /**
+     Specifies the preferred distance measurement unit.
+     - note: Anything but `kilometer` and `mile` will fall back to the default measurement for the current locale.
+        Meters and feets will be used when the presented distances are small enough. See `DistanceFormatter` for more information.
+     */
+    @objc public dynamic var distanceUnit : LengthFormatter.Unit = Locale.current.usesMetric ? .kilometer : .mile
+    
+    var usesMetric: Bool {
+        get {
+            if distanceUnit == .kilometer {
+                return true
+            } else if distanceUnit == .mile {
```

Comment:
Sorry for being stubborn, but even if we have to regress support for yards in the UK with this PR, let’s at least futureproof this code for when we do restore yard support. Replace this if/else statement with a switch statement that includes cases for meters, feet, and yards.